service/history/task/fetcher.go

```diff
@@ -104,12 +108,31 @@ func NewCrossClusterTaskFetchers(
 
 func crossClusterTaskFetchFn(
 	clientBean client.Bean,
+	sourceCluster string,
 	currentCluster string,
-	requestByShard map[int]fetchRequest,
-) (map[int]interface{}, error) {
-	// TODO: implement the fetch func after the
-	// API for fetching tasks is created.
-	return nil, errors.New("not implemented")
+	requestByShard map[int32]fetchRequest,
+) (map[int32]interface{}, error) {
+	adminClient := clientBean.GetRemoteAdminClient(sourceCluster)
+	shardIDs := make([]int32, 0, len(requestByShard))
+	for shardID := range requestByShard {
+		shardIDs = append(shardIDs, shardID)
+	}
+	request := &types.GetCrossClusterTasksRequest{
+		ShardIDs:      shardIDs,
+		TargetCluster: currentCluster,
```

Comment:
Are you planning to set the maxTaskID here sometime in the future? If we don't set, the response might be too big in case there are lots of tasks in the queue?

If you are planning to add that maybe you can add a comment here.

Reply:
My currently plan is let the source cluster decide how many tasks should be returned in one response. So if there's lots of tasks in the queue, they will be fetched in batches.

Reply:
Thanks for the explanation. Could you add this as a comment.

Reply:
Sure. Added it as comment.